be/src/storage/lake/tablet_writer.h

```diff
@@ -108,12 +124,20 @@ class TabletWriter {
     // allow to set custom tablet schema for writer, used in partial update
     void set_tablet_schema(TabletSchemaCSPtr schema) { _schema = std::move(schema); }
 
+protected:
+    // the pattern of segment_path is like this xxx/table_id/partition_id/data/segment_file_name
+    void set_segment_size(std::string segment_path, uint64_t segment_size) {
+        auto pos = segment_path.find_last_of("/");
+        _files_to_size.emplace(segment_path.substr(pos + 1), segment_size);
+    }
+
 protected:
     TabletManager* _tablet_mgr;
     int64_t _tablet_id;
     TabletSchemaCSPtr _schema;
     int64_t _txn_id;
     std::vector<std::string> _files;
+    std::unordered_map<std::string, uint64_t> _files_to_size;
```

Comment:
It's weird to use a separate map for file size, we should have a way to get both the filename and size at the same time.